clang/lib/Driver/Driver.cpp

```diff
@@ -4791,6 +4791,20 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
     if (!UnbundlerInputs.empty()) {
       Action *PartialLink =
           C.MakeAction<PartialLinkJobAction>(UnbundlerInputs, types::TY_Object);
+      if (!LastArg) {
+        auto *TA = dyn_cast<Action>(UnbundlerInputs.back());
+        assert(TA->getKind() == Action::InputClass ||
+               TA->getKind() == Action::OffloadClass);
+
+        // If the Action is of OffloadAction type, use the HostAction of the
+        // specific Offload Action to set LastArg
+        if (auto *IA = dyn_cast<OffloadAction>(UnbundlerInputs.back()))
+          LastArg =
+              &(dyn_cast<InputAction>(IA->getHostDependence())->getInputArg());
```

Comment:
For consistency:
```suggestion
        if (auto *OA = dyn_cast<OffloadAction>(UnbundlerInputs.back()))
          LastArg =
              &(dyn_cast<InputAction>(OA->getHostDependence())->getInputArg());
```

Reply:
Thank you Artem for the review, done with the suggested change.